bin/compile

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+# https://devcenter.heroku.com/articles/buildpack-api#bin-compile
+
+BUILD_DIR="${1}"
+CACHE_DIR="${2}"
+# ENV_DIR="${3}"
+
+panic() {
+    echo "FATAL: ${*}" | indent
+    exit 1
+}
+
+step() {
+    echo "-----> ${*}"
+}
+
+info() {
+    echo "${*}" | indent
+}
+
+indent() {
+    sed 's/^/       /'
+}
+
+curl_download() {
+    local url="${1}"
+    curl --proto '=https' --tlsv1.2 \
+        --silent \
+        --show-error \
+        --fail \
+        --location "${url}"
+}
+
+github_latest_tag() {
+    local owner="${1}"
+    local repo="${2}"
+    curl_download "https://api.github.com/repos/${owner}/${repo}/releases/latest" \
+        | grep --perl-regexp --only-matching '(?<="tag_name": ")(\S+)(?=")'
+}
+
+main() {
+    local latest_tag latest_version download_path layer_dir caddy_url
+
+    if [ "${CADDY_RELEASE:-}" == "" ]; then
+        step "Asking GitHub for latest version of Caddy..."
+        latest_tag="$(github_latest_tag caddyserver caddy)"
+
+        # make sure we have a version of caddy we've tested with (currently 2.x)
+        case "${latest_tag}" in
+            v2.*)
+                info "Caddy version: ${latest_tag}"
+            ;;
+            *)
+                panic "Expected caddy version 2.x -- got ${latest_tag}"
+            ;;
+        esac
```

Comment:
this will mean the builds will start crashing the day 3.x is released. What do you think about only looking at 2.x releases like this? or use this to find the latest 2.x and warn if 3.x is available?
Herokus images have jq preinstalled
```bash
curl -Ls \
-H "Accept: application/vnd.github+json" \
-H "X-GitHub-Api-Version: 2022-11-28"  \
https://api.github.com/repos/caddyserver/caddy/releases | \
jq -r 'map(select((.prerelease | not) and (.draft | not) and (.tag_name | startswith("v2")))) | max_by(.tag_name) | .tag_name'
```

Reply:
:thinking: good idea.

i do hate `jq` very very much... but if you're able to write the code for me, then sure, why not?

Reply:
now i am curious, will ask you about that hatred later today 😅 
here we go: https://github.com/thermondo/heroku-buildpack-caddy/commit/040e77b4b5a7668a445b9487874d1ddd8d0735ab

and this if you want: https://github.com/thermondo/heroku-buildpack-caddy/commit/739b0247a101f01b67aa0a22e2e6382345d2c4f3

cherry pick, redo or don't use it, however you want 🙃

Reply:
as we discussed offline: choosing an appropriate release for ourselves (reliably) involves actually parsing the version numbers. that's kind of a bit much. better to depend on the GitHub release marked as "latest."

though fwiw i improved the UX around this a bit, with a more detailed message that tells users to pin the caddy version as a temporary workaround, and check this repo for updates that support the next major version. i think that's a good compromise.